alpha_automl/builtin_primitives/semisupervised_classifier.py

```diff
@@ -0,0 +1,130 @@
+import logging
+
+import numpy as np
+import pandas as pd
+from sklearn.calibration import CalibratedClassifierCV
+from sklearn.linear_model import SGDClassifier
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.semi_supervised import (
+    LabelPropagation,
+    LabelSpreading,
+    SelfTrainingClassifier,
+)
+
+from alpha_automl.base_primitive import BasePrimitive
+from alpha_automl.utils import SemiSupervisedSplitter
+
+logger = logging.getLogger(__name__)
+
+
+class SkSelfTrainingClassifier(BasePrimitive):
+    sdg_params = dict(alpha=1e-5, penalty="l2", loss="log_loss")
+    model = SelfTrainingClassifier(SGDClassifier(**sdg_params), verbose=True)
+
+    def fit(self, X, y=None):
+        self.model.fit(X, y)
+        return
+
+    def predict(self, X):
+        pred = self.model.predict(X)
+
+        return np.array(pred)
+
+
+def make_label_pipeline(method, X):
+    step = None
+    if method == "LabelSpreading":
+        step = ("sklearn.semi_supervised.LabelSpreading", LabelSpreading())
+    elif method == "LabelPropagation":
+        step = ("sklearn.semi_supervised.LabelPropagation", LabelPropagation())
+    else:
+        return None
+
+    if isinstance(X, np.ndarray):
+        pipe = Pipeline([step])
+    else:
+        pipe = Pipeline(
+            [
+                ("toarray", FunctionTransformer(lambda x: x.toarray())),
+                step,
+            ]
+        )
+    return pipe
+
+
+class SkLabelSpreading(BasePrimitive):
+    def fit(self, X, y=None):
+        self.pipe = make_label_pipeline("LabelSpreading", X)
+        self.pipe.fit(X, y)
+        return
```

Comment:
Shouldn't this return `self`? Otherwise no need for a return statement. The same goes for `SkLabelPropagation.fit` below.